src/WaveTrack.h

```diff
@@ -148,7 +148,8 @@ class AUDACITY_DLL_API WaveTrack final : public PlayableTrack {
    int GetWaveColorIndex() const { return mWaveColorIndex; };
    void SetWaveColorIndex(int colorIndex);
 
-   sampleCount GetNumSamples() const;
+   sampleCount GetPlaySamplesCount() const;
+   sampleCount GetSequenceSamplesCount() const;
```

Comment:
I see that WaveTrack::GetNumSamples becomes two public methods, each used in only one other place, and those other uses are correct.

I like the renaming changes here and in WaveClip -- *Sample* versus *Play* names -- and they force you to recompile many places and consider carefully which renaming is the correct one.  I am more confident in your work when I see that you did this.  Should the name changes be permanent?  I say, yes.

Much less has changed in the interface of WaveTrack.h, than for WaveClip.h.  It is a little surprising -- so to review carefully I must also look at what else is in WaveTrack.h that did not change, and be convinced that it is correct not to change them.

I notice that WaveTrack::GetStartTime() and GetEndTime() did not change names.  But as you know, that is because it is an override of a method of Track.h, which declares virtual GetStartTime() and GetEndTime() but doesn't comment what they are supposed to mean.  So, I understand this exception to renaming -- you don't want to make the name changes in the other overrides.

Still that means I must review all uses of GetStartTime and GetEndTime to be sure.

Reply:
I'm doing that review, and I notice many GetStartTime and GetEndTime calls in effects, which I just assume correct -- that destructively applied affects will only apply to the un-hidden parts of tracks.  As I said, that might be surprising in some cases and a debatable point of design, but I will interpret the design for now as not demanding it.

I notice GetOffset() and SetOffset() are two other virtual functions in the base class Track that were not renamed, so I must also review uses of those to be sure of completeness.

Reply:
Now I have done that review.  I think that the legacy in class Track needs a clean-up.  It's not clear what the distinction of "start time" versus "offset" is supposed to mean, or what purpose the mOffset member variable serves.  I think there should be a clean-up project to eliminate the "Offset" functions.

This is not to be part of the present pull request.  It is for one of us to write in the stabilization period and for the other one to review.